packages/admin/cms-admin/src/form/file/FinalFormFileUpload.tsx

```diff
@@ -0,0 +1,175 @@
+            onDrop={async (acceptedFiles, rejectedFiles) => {
+                setFailedUploads([]);
+
+                const tooManyFilesWereDropped = rejectedFiles.some((rejection) => rejection.errors.some((error) => error.code === "too-many-files"));
+                setTooManyFilesSelected(tooManyFilesWereDropped);
+
+                rejectedFiles.map((rejection) => {
+                    const failedFile: ErrorFileSelectItem = {
+                        name: rejection.file.name,
+                        error: true,
+                    };
+
+                    if (rejection.errors.some((error) => error.code === "file-too-large")) {
+                        failedFile.error = <FormattedMessage id="comet.finalFormFileUpload.fileTooLarge" defaultMessage="File is too large." />;
+                    }
+
+                    setFailedUploads((existing) => [...existing, failedFile]);
+                });
+
+                if (singleFile) {
+                    input.onChange(undefined);
+                }
+
+                if (tooManyFilesWereDropped || !acceptedFiles.length) {
+                    return;
+                }
+
+                setUploadingFiles(acceptedFiles.map((file) => ({ name: file.name, loading: true })));
+
+                const fetches = acceptedFiles.map((file) => {
+                    const formData = new FormData();
+                    formData.append("file", file);
+                    return fetch(`${apiUrl}/public-upload/files/upload`, {
+                        method: "POST",
+                        body: formData,
+                    });
+                });
+
+                const responses = await Promise.all(fetches)
+                    .then((responses) => Promise.all(responses.map((response) => response.json() as Promise<ApiResponse>)))
+                    .finally(() => {
+                        setUploadingFiles(uploadingFiles.filter((loadingFile) => !acceptedFiles.some((file) => file.name === loadingFile.name)));
+                    });
+
+                const successfulUploadResponses = responses.filter((item) => "id" in item) as SuccessfulApiResponse[];
+                const failedUploads = acceptedFiles.filter((file) => !successfulUploadResponses.some((item) => item.name === file.name));
+
+                failedUploads.forEach((file) => {
+                    setFailedUploads((prevFailedUploads) => [
+                        ...prevFailedUploads,
+                        {
+                            name: file.name,
+                            error: <FormattedMessage id="comet.finalFormFileUpload.uploadFailed" defaultMessage="Upload failed." />,
+                        },
+                    ]);
+                });
+
+                const successfullyUploadedFiles = [...successfulUploadResponses.map(({ id, name, size }) => ({ id, name, size }))];
+
+                if (singleFile) {
+                    if (successfullyUploadedFiles.length) {
```

Comment:
```suggestion
                    if (successfullyUploadedFiles.length > 0) {
```

Reply:
This shouldn't have any logical impact and I don't think adding `> 0` makes it more readable/understandable 🤔

Reply:
IMO it is more readable/understandable because I don't need to remember that 0 is falsy and anything other than 0 is truthy. But I don't like falsy/truthy checks in general 🤷🏼‍♂️